### PR TITLE
Add search for ExpenseCategory to NetSuite Records

### PR DESCRIPTION
**Pull request by azellsworth**
- pulls in all categories from NetSuite

@ahuth @shirish-pampoorickal @AnnaMhairi

### CODE REVIEW

File: lib/netsuite/records/expense_category.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module NetSuite
+  module Records
+    class ExpenseCategory
+      include Support::Fields
+      include Support::Actions
+      include Namespaces::ListAcct
+
+      actions :search
```
Comment (shirish-pampoorickal) [inline lib/netsuite/records/expense_category.rb:10]: Can you add the reference link to be consistent with other record files?
https://system.netsuite.com/help/helpcenter/en_US/srbrowser/Browser2014_1/script/record/expensecategory.html

Comment (shirish-pampoorickal) [inline lib/netsuite/records/expense_category.rb:10]: Is there a reason we are not adding the standard actions?
```
actions :get, :get_list, :add, :delete, :search, :update, :upsert
```

File: lib/netsuite/records/expense_category.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+      fields :name
+
+      attr_reader :internal_id
+
+      def initialize(attributes = {})
+        @internal_id = attributes.delete(:internal_id) || attributes.delete(:@internal_id)
+        initialize_from_attributes_hash(attributes)
+      end
+    end
```
Comment (shirish-pampoorickal) [inline lib/netsuite/records/expense_category.rb:24]: Also, seems like any record with search action has the follow methods:
```
     # Copied over from lib/netsuite/records/expense_report.rb

      # seems like this is necessary
      def self.search_class_name
        "Transaction"
      end

      # not sure if this is needed
      def self.search_class_namespace
        "tranSales"
      end
```

**Review by shirish-pampoorickal — CHANGES_REQUESTED** [review]
I think we are missing a few things

File: lib/netsuite/records/expense_category.rb (hunk 1 above)
Comment (azellsworth) [reply lib/netsuite/records/expense_category.rb:10]: We only know the search works at this point... I guess fine to add the others, we just haven't made sure they worked.

Comment (azellsworth) [reply lib/netsuite/records/expense_category.rb:10]: Done

**Review by ahuth — APPROVED** [review]

File: lib/netsuite/records/expense_category.rb (hunk 2 above)
Comment (azellsworth) [reply lib/netsuite/records/expense_category.rb:24]: Done

**Comment by azellsworth** [comment]
@shirish-pampoorickal comments addressed

**Review by shirish-pampoorickal — APPROVED** [review]
We should look into adding the missing standard actions in a later pull request. Other than that Code LGTM!

**Pull request merged**